bin/cylc-get-host-metrics

```diff
@@ -0,0 +1,214 @@
+     -/+ buffers/cache:    3050188  *3066664*
+     Swap:      4190204      14624    4175580
+
+   DISK_TEMPLATES, e.g. for mount directory path option '--disk-space=/boot'
+     $ df -Pk /boot
+     Filesystem            1024-blocks      Used Available Capacity Mounted on
+     /dev/md0                    96978     33904   *57856*      37% /boot."""
+LOAD_TEMPLATES = ['uptime', (1, 5, 15), re.compile(
+                  r'load average:\s(\d+\.?\d+),\s(\d+\.?\d+),\s(\d+\.?\d+)')]
+MEMORY_TEMPLATES = ['free -k', re.compile(
+                    r'\-/\+ buffers/cache:\s+\d+\s+(\d+)')]
+DISK_TEMPLATES = ['df -Pk', re.compile(
+                  r'Mounted on\n\S+\s+\d+\s+\d+\s+(\d+)\s+')]
+
+
+def main():
+    parser = COP(__doc__, comms=True, noforce=True, argdoc=[])
+
+    parser.add_option(
+        "--load", "-l",
+        help="1, 5 and 15 minute load averages from the 'uptime' command",
+        action="store_true", default=False, dest="load")
+
+    parser.add_option(
+        "--memory", "-m",
+        help="Total free RAM memory, in KB, from the 'free -k' command.",
+        action="store_true", default=False, dest="memory")
+
+    parser.add_option(
+        "--disk-space",
+        help="Available disk space, in KB, from the 'df -Pk' command",
+        action="store", default=None, dest="disk")
+
+    options, args = parser.parse_args()
+    print get_host_metric(options.load, options.memory, options.disk)
+
+    sys.exit()
+
+
+def run_command_string(command_str):
+    """Return the result from running a command string in a local shell.
+
+       Return stdout if command successful, else raise CalledProcessError."""
+    # Inputs to process are internal only, so 'shell=True' is safe here.
+    process = Popen(command_str, shell=True, stdin=open(os.devnull),
+                    stdout=PIPE, stderr=PIPE)
+    stdoutput = process.communicate()[0]
+    process.wait()
+    return_code = process.returncode
+    if return_code:
+        raise CalledProcessError(return_code, command_str)
+    else:
+        return str(stdoutput)
+
+
+def extract_pattern(data_pattern, raw_string):
+    """Try to return first parenthesized subgroup of a regex string search."""
+    error_msg = "No 're.search' matches for:\n  %s\non the string:\n  %s" % (
+        data_pattern.pattern, raw_string)
+    try:
+        matches = re.search(data_pattern, raw_string)
+    except re.error:
+        sys.stderr.write(error_msg)
+    if matches:
+        return matches.groups()
+    else:
+        raise AttributeError(error_msg)
+
+
+def process_load():
+    """Return dictionary with load average float values from 'uptime' command.
+
+       String keys e.g. 'load:1' give minutes average was measured over."""
+    cmd, av_times, patt = LOAD_TEMPLATES
+    av_raw_tuple = extract_pattern(patt, run_command_string(cmd))
+    try:
+        av_floats_tuple = [float(av_raw) for av_raw in av_raw_tuple]
+    except TypeError:
```

Comment:
If we cant cast the memory as a float can we safely continue?

Reply:
Indeed, no. Though if there was an error converting to float, this would ultimately indicate an issue with the regular expression handling. I'll have another think about how to handle errors relating to the data extraction.

Reply:
I've changed the cases as highlighted to simply 'raise' the associated exception, which it turns out would be a ``ValueError`` not a ``TypeError`` (oops!)